File: CS/WinFormsDashboard_RecalculateTotals/Classes/DashboardTotalsHelper.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardCommon.Localization;
using DevExpress.Data;

namespace WinFormsDashboard_RecalculateTotals {
    public class DashboardTotalsHelper {
        public static SummaryItemType GetGridColumnTotalType(GridColumnTotalType dashboardTotalType) {
            switch (dashboardTotalType) {
                case GridColumnTotalType.Sum: return SummaryItemType.Sum;
                case GridColumnTotalType.Avg: return SummaryItemType.Average;
                case GridColumnTotalType.Count: return SummaryItemType.Count;
                case GridColumnTotalType.Max: return SummaryItemType.Max;
                case GridColumnTotalType.Min: return SummaryItemType.Min;
                default: return SummaryItemType.Custom;
            };
        }
        public static string GetTotalPrefix(GridColumnTotalType dashboardTotalType) {
            switch (dashboardTotalType) {
                case GridColumnTotalType.Sum: return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeSum);
                case GridColumnTotalType.Avg: return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeAvg);
                case GridColumnTotalType.Count: return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeCount);
                case GridColumnTotalType.Max: return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeMax);
                case GridColumnTotalType.Min: return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeMin);
                default: return null;
            };
        }
    }
}

File: VB/WinFormsDashboard_RecalculateTotals/Classes/DashboardTotalsHelper.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardCommon.Localization
Imports DevExpress.Data

Namespace WinFormsDashboard_RecalculateTotals

    Public Class DashboardTotalsHelper

        Public Shared Function GetGridColumnTotalType(ByVal dashboardTotalType As GridColumnTotalType) As SummaryItemType
            Select Case dashboardTotalType
                Case GridColumnTotalType.Sum
                    Return SummaryItemType.Sum
                Case GridColumnTotalType.Avg
                    Return SummaryItemType.Average
                Case GridColumnTotalType.Count
                    Return SummaryItemType.Count
                Case GridColumnTotalType.Max
                    Return SummaryItemType.Max
                Case GridColumnTotalType.Min
                    Return SummaryItemType.Min
                Case Else
                    Return SummaryItemType.Custom
            End Select
        End Function

        Public Shared Function GetTotalPrefix(ByVal dashboardTotalType As GridColumnTotalType) As String
            Select Case dashboardTotalType
                Case GridColumnTotalType.Sum
                    Return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeSum)
                Case GridColumnTotalType.Avg
                    Return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeAvg)
                Case GridColumnTotalType.Count
                    Return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeCount)
                Case GridColumnTotalType.Max
                    Return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeMax)
                Case GridColumnTotalType.Min
                    Return DashboardLocalizer.GetString(DashboardStringId.GridTotalTypeMin)
                Case Else
                    Return Nothing
            End Select
        End Function
    End Class
End Namespace
